File: backend-dotnet/Controllers/WishlistController.cs
using Microsoft.AspNetCore.Mvc;
using ClothingStore.Services;
using ClothingStore.Models;

namespace ClothingStore.Controllers
{
    [ApiController]
    [Route("api/wishlist")]
    public class WishlistController : ControllerBase
    {
        private readonly WishlistService _wishlistService;
        
        public WishlistController(WishlistService wishlistService)
        {
            _wishlistService = wishlistService;
        }
        
        [HttpPost("add")]
        public async Task<ActionResult> AddToWishlist([FromBody] WishlistRequest request)
        {
            try
            {
                var message = await _wishlistService.AddToWishlistAsync(request.UserId, request.ProductId);
                return Ok(new { message });
            }
            catch (Exception e)
            {
                return BadRequest(new { error = e.Message });
            }
        }
        
        [HttpGet("user/{userId}")]
        public async Task<ActionResult<List<Wishlist>>> GetWishlist(long userId)
        {
            var wishlist = await _wishlistService.GetWishlistAsync(userId);
            return Ok(wishlist);
        }
        
        [HttpGet("check")]
        public async Task<ActionResult<bool>> IsInWishlist([FromQuery] long userId, [FromQuery] long productId)
        {
            var isInWishlist = await _wishlistService.IsInWishlistAsync(userId, productId);
            return Ok(isInWishlist);
        }
        
        [HttpDelete("remove")]
        public async Task<ActionResult> RemoveFromWishlist([FromQuery] long userId, [FromQuery] long productId)
        {
            try
            {
                await _wishlistService.RemoveFromWishlistAsync(userId, productId);
                return Ok(new { message = "Removed from wishlist" });
            }
            catch (Exception e)
            {
                return BadRequest(new { error = e.Message });
            }
        }
    }
    
    public class WishlistRequest
    {
        public long UserId { get; set; }
        public long ProductId { get; set; }
    }
}

File: genaibackend/src/main/java/com/clothingstore/controller/WishlistController.java
package com.clothingstore.controller;

import com.clothingstore.entity.Wishlist;
import com.clothingstore.repository.WishlistRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.transaction.annotation.Transactional;
import org.springframework.web.bind.annotation.*;

import java.util.List;
import java.util.Map;

@RestController
@RequestMapping("/api/wishlist")
@CrossOrigin(origins = "*")
public class WishlistController {
    
    @Autowired
    private WishlistRepository wishlistRepository;
    
    @PostMapping("/add")
    public ResponseEntity<?> addToWishlist(@RequestBody Map<String, Object> request) {
        try {
            Long userId = Long.valueOf(request.get("userId").toString());
            Long productId = Long.valueOf(request.get("productId").toString());
            
            if (wishlistRepository.findByUserIdAndProductId(userId, productId).isPresent()) {
                return ResponseEntity.ok(Map.of("message", "Already in wishlist"));
            }
            
            Wishlist wishlist = new Wishlist(userId, productId);
            wishlistRepository.save(wishlist);
            return ResponseEntity.ok(Map.of("message", "Added to wishlist"));
        } catch (Exception e) {
            return ResponseEntity.badRequest().body(Map.of("error", e.getMessage()));
        }
    }
    
    @GetMapping("/user/{userId}")
    public ResponseEntity<List<Wishlist>> getWishlist(@PathVariable Long userId) {
        return ResponseEntity.ok(wishlistRepository.findByUserId(userId));
    }
    
    @DeleteMapping("/remove")
    @Transactional
    public ResponseEntity<?> removeFromWishlist(@RequestParam Long userId, @RequestParam Long productId) {
        try {
            wishlistRepository.deleteByUserIdAndProductId(userId, productId);
            return ResponseEntity.ok(Map.of("message", "Removed from wishlist"));
        } catch (Exception e) {
            return ResponseEntity.badRequest().body(Map.of("error", e.getMessage()));
        }
    }
}